expressjs/src/routes/v1/System.ts

```diff
@@ -22,479 +24,33 @@
-        res.json({ data })
-      } catch (error) {
-        Logger.error(error)
-        res.status(500)
-        res.json({ error })
-      }
-    } else if (reqBody.cmd === '6') {
-      try {
-        const data = await si.dockerInfo()
-        res.json({ data })
-      } catch (error) {
-        Logger.error(error)
-        res.status(500)
-        res.json({ error })
-      }
-    } else if (reqBody.cmd === '7') {
-      try {
-        const data = await si.dockerImages()
-        res.json({ data })
-      } catch (error) {
-        Logger.error(error)
-        res.status(500)
-        res.json({ error })
-      }
-    } else if (reqBody.cmd === '8') {
-      try {
-        const data = await si.dockerContainers(true)
-        res.json({ data })
-      } catch (error) {
-        Logger.error(error)
-        res.status(500)
-        res.json({ error })
-      }
-    } else if (reqBody.cmd === '9') {
-      try {
-        const data = await si.dockerContainerStats('*')
-        res.json({ data })
-      } catch (error) {
-        Logger.error(error)
-        res.status(500)
-        res.json({ error })
-      }
-    } else if (reqBody.cmd === '0') {
-      try {
-        const data = await si.dockerContainerProcesses('*')
-        res.json({ data })
-      } catch (error) {
-        Logger.error(error)
-        res.status(500)
-        res.json({ error })
-      }
-    } else if (reqBody.cmd === '+') {
-      try {
-        const data = await si.dockerVolumes()
-        res.json({ data })
-      } catch (error) {
-        Logger.error(error)
-        res.status(500)
-        res.json({ error })
-      }
-    } else if (reqBody.cmd === ',') {
-      try {
-        const data = await si.getStaticData()
-        res.json({ data })
-      } catch (error) {
-        Logger.error(error)
-        res.status(500)
-        res.json({ error })
-      }
-    } else if (reqBody.cmd === '.') {
-      try {
-        const data = (await si.getDynamicData(
-          'apache2, postgres, wsearch'
-        )) as unknown
-        res.json({ data })
-      } catch (error) {
-        Logger.error(error)
-        res.status(500)
-        res.json({ error })
-      }
-    } else if (reqBody.cmd === '/') {
-      try {
-        const data = (await si.getAllData(
-          'apache2, postgres, wsearch'
-        )) as unknown
-        res.json({ data })
-      } catch (error) {
-        Logger.error(error)
-        res.status(500)
-        res.json({ error })
-      }
+    try {
+      const reqBody = req.body as reqBodyData
+
+      const data = (await getSiFunction(reqBody.cmd)) as JSON
+      res.json({ data })
+    } catch (error) {
+      Logger.error(error)
+      res.status(500)
+      res.json({ error })
     }
   }
 )
 
+// Functions
+function getSiFunction(funcName: string) {
+  if (si[funcName as keyof typeof si]) {
+    // Disable warnings about using `any`
+    // eslint-disable-next-line
+    return (si as any)[funcName]()
```

Comment:
This change allows us to use one singe source of truth for the System Information package. It allows the frontend to pass the name of the SystemInformation function it wants to call, rather than having to create a bunch of switch cases and maintain that list as well as an additional list on the frontend for the selector on the system info page. This means that someone could technically send an API request and call any of the `systeminformation` functions at will by passing the name of the function. These should all be harmless functions that merely fetch system data like CPU stats and therefore low risk.

May consider some alternatives though, I don't want to have to remember to be constantly monitoring the systeminformation package to ensure no riskier functions are added.